src/main/java/org/folio/dew/batch/bulkedit/jobs/BulkEditHoldingsProcessor.java

```diff
@@ -126,38 +122,36 @@ private ExtendedHoldingsRecordCollection getHoldingsRecords(ItemIdentifier itemI
         if (INSTANCEHRID != identifierTypeEnum && tenantIds.size() > 1) {
           throw new BulkEditException(DUPLICATES_ACROSS_TENANTS);
         }
-        tenantIds.forEach(tenantId -> {
+        var affiliatedPermittedTenants = tenantResolver.getAffiliatedPermittedTenantIds(EntityType.HOLDINGS_RECORD,
+          jobExecution, identifierType, tenantIds, itemIdentifier);
+        affiliatedPermittedTenants.forEach(tenantId -> {
           try (var context = new FolioExecutionContextSetter(refreshAndGetFolioExecutionContext(tenantId, folioExecutionContext))) {
-            permissionsValidator.checkBulkEditReadPermissions(tenantId, EntityType.HOLDINGS_RECORD);
             var holdingsRecordCollection = getHoldingsRecordCollection(type, itemIdentifier);
             extendedHoldingsRecordCollection.getExtendedHoldingsRecords().addAll(
               holdingsRecordCollection.getHoldingsRecords().stream()
                 .map(holdingsRecord -> new ExtendedHoldingsRecord().tenantId(tenantId).entity(holdingsRecord)).toList()
             );
             extendedHoldingsRecordCollection.setTotalRecords(extendedHoldingsRecordCollection.getTotalRecords() + holdingsRecordCollection.getTotalRecords());
           } catch (Exception e) {
-            if (e instanceof FeignException && ((FeignException) e).status() == 401) {
```

Comment:
Do we really no longer need this logic, and should we not handle errors related to affiliations and permissions separately? This story is related to adding the Eureka integration but not to changing the integration with Okapi.

Reply:
Hi @khandramai , this logic was moved to tenantResolver.getAffiliatedPermittedTenantIds in the way to check all needed tenants - related to permissions and affiliations requirements.